### PR TITLE
Update completions.rs

### PR DESCRIPTION
**Pull request by kaksv**
Adding the closing parenthesis on line 285

- - -
This change is [Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5966)

### CODE REVIEW

**Review by mkaput — CHANGES_REQUESTED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5966)* status: all files reviewed, 1 unresolved discussion (waiting on @kaksv)

___
*[`crates/cairo-lang-language-server/src/ide/completion/completions.rs` line 285 at r1](https://reviewable.io/reviews/starkware-libs/cairo/5966#-O0z59gM1c-DaGaz23Kz:-O0z59gM1c-DaGaz23L-:b397l1f) ([raw file](https://github.com/starkware-libs/cairo/blob/ea77b086127603b5bb7f4d5efc99d96a5459caa6/crates/cairo-lang-language-server/src/ide/completion/completions.rs#L285)):*
> ```XML
>     let completion = CompletionItem {
>         label: format!("{}()", name),
>         insert_text: Some(format!("{}(", name))),
> ```

This change breaks syntax?

File: crates/cairo-lang-language-server/src/ide/completion/completions.rs [hunk 1]
```diff
@@ -282,7 +282,7 @@ fn completion_for_method(
 
     let completion = CompletionItem {
         label: format!("{}()", name),
-        insert_text: Some(format!("{}(", name)),
+        insert_text: Some(format!("{}(", name))),
```
Comment (LucasLvy) [inline crates/cairo-lang-language-server/src/ide/completion/completions.rs:285]: Is that what you meant ?
```suggestion
        insert_text: Some(format!("{}()", name)),
```

Comment (kaksv) [reply crates/cairo-lang-language-server/src/ide/completion/completions.rs:285]: Yes, because the statement gets to fully close.

**Review by mkaput — CHANGES_REQUESTED** [review]
Please notify me if you want to address my comments or if I should close this PR and put the issue back for taking :)

> Reviewed 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/starkware-libs/cairo/5966#-:-O3DiF26BH1z2rrX3clF:bs64pok)* status: all files reviewed, 3 unresolved discussions (waiting on @0xLucqs and @kaksv)

*[a discussion](https://reviewable.io/reviews/starkware-libs/cairo/5966#-O3Di-WEEurfxIXHmTeV:-O3Di-WEEurfxIXHmTeW:b1jzgeb) (no related file):*
Please add tests for this change.

We do not have any completions tests yet, but they should be pretty trivial to add because we have very similar tests for hovers already.

Hover tests are coded here:
https://github.com/starkware-libs/cairo/blob/00c1c18ece73a5b503db8998eb81fed45a7e8b3d/crates/cairo-lang-language-server/tests/e2e/hover.rs#L11

and they are using snapshot fixtures, example one is here:
https://github.com/starkware-libs/cairo/blob/4646ad819521338ccd92b8ebef76b96770b46ce1/crates/cairo-lang-language-server/tests/test_data/hover/basic.txt#L1

The test plan I imagine should look like the issue description I did, i.e. using the `cursors` mechanism you form a situation of the code before completion, and a situation after it. Then you run the completion request over the before snippet, apply the result and you compare what you got.

Here you have a ton of examples of such tests from IntelliJ Rust codebase: https://github.com/intellij-rust/intellij-rust/blob/c6657c02bb62075bf7b7ceb84d000f93dda34dc1/src/test/kotlin/org/rust/lang/core/completion/RsCompletionTest.kt

___
___
*[`crates/cairo-lang-language-server/src/ide/completion/completions.rs` line 285 at r2](https://reviewable.io/reviews/starkware-libs/cairo/5966#-O3DiAIm0hG42mfSMr8l:-O3DiAIm0hG42mfSMr8m:bipcda9) ([raw file](https://github.com/starkware-libs/cairo/blob/8e92a1aebf73a76b6e6d3d695e026aed4c13d75f/crates/cairo-lang-language-server/src/ide/completion/completions.rs#L285)):*
> ```XML
>     let completion = CompletionItem {
>         label: format!("{}()", name),
>         insert_text: Some(format!("{}()", name)),
> ```

This sounds like it always puts the cursor after `)`, isn't it?

Can you put the cursor before `)` if the completed function has any arguments?

See this video on how it's done in RustRover:

[Nagranie z ekranu 2024-07-4 o 10.53.56.mov](https://files.reviewableusercontent.io/images/reviewable/3450050/qN88uoXB1DNdjfi3kscuxgYP/Nagranie%20z%20ekranu%202024-07-4%20o%2010.53.56.mov)

**Comment by mkaput** [comment]
closing due to lack of activity

**Pull request closed without merging**